component/block_cache/block_cache_test.go

```diff
@@ -159,11 +160,11 @@ func (suite *blockCacheTestSuite) TestEmpty() {
 	suite.assert.Nil(err)
 	suite.assert.Equal(tobj.blockCache.Name(), "block_cache")
 	suite.assert.EqualValues(tobj.blockCache.blockSize, 16*_1MB)
-	suite.assert.EqualValues(tobj.blockCache.memSize, 4192*_1MB)
-	suite.assert.EqualValues(tobj.blockCache.diskSize, 4192)
+	//Removed memory check since memory calculated during testing may differ
+	suite.assert.EqualValues(tobj.blockCache.diskSize, 0)
 	suite.assert.EqualValues(tobj.blockCache.diskTimeout, defaultTimeout)
-	suite.assert.EqualValues(tobj.blockCache.workers, 128)
-	suite.assert.EqualValues(tobj.blockCache.prefetch, MIN_PREFETCH)
+	suite.assert.EqualValues(tobj.blockCache.workers, uint32(3*runtime.NumCPU()))
+	suite.assert.EqualValues(tobj.blockCache.prefetch, uint32(2*runtime.NumCPU()))
```

Comment:
we shall check for disk and memory size also here for autoconfiguration

Reply:
I added a memory check but 2/3 times it showed the test as failed because it wasn't able to capture the exact same value for free memory which the code had captured before while calculating.

Reply:
Another thing for this test case the config is empty with contents read-only: true. Since tmp-path is not configured and disk-size-mb is not given the disk-size is not set and value is 0. So, I am validating that here.